### PR TITLE
Update outdated docs

### PR DESCRIPTION
**Pull request by anselmbradford**
Updates outdated docs and removes the "what's new" doc (fixes https://github.com/cfpb/cfgov-refresh/issues/3697)

### CODE REVIEW

File: docs/development-tips.md [hunk 1]
```diff
@@ -3,27 +3,16 @@
...
 but are not fully incorporated into the project.
-These are known as "non-v1 Django apps."
-In order to visit areas of the site locally where those projects are used,
-the sibling projects need to be installed
-and then indexed within this cfgov-refresh project.
+These are known as "satellite apps."
 
-The non-v1 apps are the following:
+Satellite apps are listed in the
+[optional-public.txt](https://github.com/cfpb/cfgov-refresh/blob/master/requirements/optional-public.txt).
```
Comment (chosak) [inline docs/development-tips.md]: Extra period after the link.

File: docs/development-tips.md [hunk 2]
```diff
@@ -34,6 +23,11 @@ After installing these projects as sibling directories to the `cfgov-refresh` re
     Clone and install the projects as siblings to `cfgov-refresh`,
     so that they share the same parent directory (`~/Projects` or similar).
 
+In addition to the aforementioned list,
```
Comment (chosak) [inline docs/development-tips.md]: Maybe move this up near the line about `optional-public`? Encountering this at the end is a bit unclear as these apps aren't included in that requirements file.

File: docs/development-tips.md (hunk 1 above)
Comment (anselmbradford) [reply docs/development-tips.md]: The period is grammatically correct, isn't it?

Comment (anselmbradford) [reply docs/development-tips.md]: Oh snap, nvm, I see what you mean.

**Comment by chosak** [comment]
@anselmbradford thanks! There are two minor typos in there (should be "Apps" not "App's" for the last two) but looks great.

**Comment by anselmbradford** [comment]
@chosak cool, pushed up https://github.com/cfpb/cfgov-refresh/pull/3873/commits/59c30baabf8d6051f67759e9f5aafa2be8a61d1a

**Pull request merged**